<?xml version="1.0" ?>
<sdf version="1.5">
  <model name="vr_view_vive_pro">
    <pose>0 0 0 0 0 0</pose>



    <link name="left_eye">
      <pose>0 0.0328 0 0 0 0</pose>
      <inertial>
        <mass>0</mass>
      </inertial>
  
 <visual name="visual">
      <pose>0 0 0 0 1.57075 0</pose>
        <geometry>
          <cylinder>
            <radius>0.05</radius>
            <length>0.05</length>
          </cylinder>
        </geometry>
</visual>


      <sensor name="camera" type="camera">
        <camera>
          <horizontal_fov>1.347</horizontal_fov>
          <image>
            <width>1852</width>
            <height>2056</height>
          </image>
          <clip>
            <near>0.1</near>
            <far>100</far>
          </clip>
        </camera>

         <plugin name="gazebo_ros_camera" filename="libgazebo_ros_camera.so">
         <cameraName>camera</cameraName>
         <alwaysOn>true</alwaysOn>
         <updateRate>0</updateRate>
         <imageTopicName>rgb/left_eye</imageTopicName>
         <cameraInfoTopicName>rgb/left_info</cameraInfoTopicName>
         <frameName>camera_frame</frameName>
         <baseline>0.1</baseline>
         <distortion_k1>0.0</distortion_k1>
         <distortion_k2>0.0</distortion_k2>
         <distortion_k3>0.0</distortion_k3>
         <distortion_t1>0.0</distortion_t1>
         <distortion_t2>0.0</distortion_t2>
         </plugin>
        <always_on>1</always_on>
        <update_rate>40</update_rate>
      </sensor>
    </link>



    <link name="right_eye">
      <pose>0 -0.0328 0 0 0 0</pose>
      <inertial>
        <mass>0</mass>
      </inertial>
 
 <visual name="visual">
      <pose>0 0 0 0 1.57075 0</pose>
        <geometry>
          <cylinder>
            <radius>0.05</radius>
            <length>0.05</length>
          </cylinder>
        </geometry>
</visual>



      <sensor name="camera" type="camera">
        <camera>
          <horizontal_fov>1.347</horizontal_fov>
          <image>
            <width>1852</width>
            <height>2056</height>
          </image>
          <clip>
            <near>0.1</near>
            <far>100</far>
          </clip>
        </camera>

         <plugin name="gazebo_ros_camera" filename="libgazebo_ros_camera.so">
         <cameraName>camera</cameraName>
         <alwaysOn>true</alwaysOn>
         <updateRate>0</updateRate>
         <imageTopicName>rgb/right_eye</imageTopicName>
         <cameraInfoTopicName>rgb/right_info</cameraInfoTopicName>
         <frameName>camera_frame</frameName>
         <baseline>0.1</baseline>
         <distortion_k1>0.0</distortion_k1>
         <distortion_k2>0.0</distortion_k2>
         <distortion_k3>0.0</distortion_k3>
         <distortion_t1>0.0</distortion_t1>
         <distortion_t2>0.0</distortion_t2>
         </plugin>
        <always_on>1</always_on>
        <update_rate>40</update_rate>
      </sensor>
    </link>

  </model>
</sdf>
